src/engraving/libmscore/measure.cpp

```diff
@@ -3660,6 +3591,34 @@ qreal Measure::basicWidth() const
     return w;
 }
 
+//---------------------------------------------------------
+//   stretchWeight
+//   Returns a weight parameter proportional to the
+//   current with of the measure.
+//---------------------------------------------------------
+qreal Measure::stretchWeight()
+{
+    if (isMMRest()) {
+        return width();
+    }
+    qreal w = 0;
+    qreal W = 0;
+    qreal weight = 0;
+    qreal minNoteSpace = score()->noteHeadWidth() * 1.05;
+    Fraction minTicks = minSysTicks();
+    for (Segment* s = first(); s; s = s->next()) {
+        if (s->enabled() && s->visible() && !s->allElementsInvisible() && s->isChordRestType()) {
+            qreal str = stretchFormula(s->ticks(), minTicks, 1);
+            w += minNoteSpace * str * basicStretch();
+            W += s->width();
+            // You cannot simply do weight += s->width(), because the s->width() also includes accidentals
+        }
+        continue;
+    }
+    weight = w * (w / W); // Empirical correction factor (see documentation)
```

Comment:
which documentation? You're talking about PDF?  If so, then put a relevant comment here with an explanation of what the 'Empirical correction factor' is